src/components/Header.test.tsx

```diff
@@ -0,0 +1,163 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import { MemoryRouter } from 'react-router-dom'
+import { useFirebase, isLoaded, isEmpty } from 'react-redux-firebase'
+import { Provider } from 'react-redux'
+import configureStore from 'redux-mock-store'
+import { Header } from './Header'
+
+jest.mock('react-redux-firebase')
+
+let mockedUseNavigate: jest.Mock
+beforeEach(() => {
+  mockedUseNavigate = jest.fn()
+})
+jest.mock('react-router-dom', () => ({
+  ...(jest.requireActual('react-router-dom') as object),
+  useNavigate: () => mockedUseNavigate,
+}))
+
+const mockStore = configureStore()
+const initialState = { firebase: {} }
+const loggedInState = {
+  firebase: {
+    auth: 'user',
+  },
+}
+
+const mockLoggedUser = () => {
+  ;(isLoaded as unknown as jest.Mock).mockImplementation(() => true)
+  ;(isEmpty as unknown as jest.Mock).mockImplementation(() => false)
```

Comment:
These semicolons at the beginning looks so ugly 🙈  But that's my opinion, I promise to not pay attention to that in the future 😛